core/components/atoms/heading/Heading.tsx

```diff
@@ -28,30 +28,33 @@ const sizeMap = {
   l: 'h3',
   xl: 'h2',
   xxl: 'h1',
-  default: 'h3'
 };
 
 export const Heading = (props: HeadingProps) => {
   const {
-    size = 'default',
     appearance = 'default',
+    size,
     children,
     ...rest
   } = props;
 
   const classes = classNames({
     Heading: true,
-    [`Heading--${size}`]: size !== 'default',
+    [`Heading--${size}`]: size,
     [`Heading--${appearance}`]: appearance
   });
 
   return (
-    <GenericText className={classes} componentType={sizeMap[size]} {...rest}>
+    <GenericText className={classes} componentType={size ? sizeMap[size] : 'h5'} {...rest}>
```

Comment:
Use default prop for the size as `l` which will map to `h3` as it is the most common heading size used, no need to check for size and returning `h5` from if block.